site/_posts/2023-03-09-react-example-application.md

```diff
@@ -0,0 +1,247 @@
+We are not using the pre-built button components from the React SDK. Instead, we are using the `useFusionAuth` hook to get the `login` and `register` methods from the `FusionAuthContext`. This allows us to customize the login / register button.
+
+```tsx
+  const {isAuthenticated, isLoading, login, register} = useFusionAuth();
+
+  return (
+    <div className="flex flex-col w-full border-opacity-50">
+      <button onClick={() => login()} className="btn btn-primary">Login</button>
+      <div className="divider">OR</div>
+      <button onClick={() => register()} className="btn btn-primary">Register Now</button>
+    </div>
+  )
+```
+
+### Tickets Page
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-tickets.png" alt="Overview of the tickets when the user is authenticated" class="img-fluid" figure=false %}
+
+The tickets page displays the tickets of the user. The tickets are retrieved from the backend using the `/api/tickets` endpoint. The returned tickets depend on the role of the user. If the user is only a `customer`, only the tickets of the user are returned.
+
+We also embed the user information in the backend response. This allows us to display the creator information of the ticket (full name, picture).
+
+```tsx
+<div className="flex items-center space-x-3">
+  <Avatar name={ticket._creator.firstName + ' ' + ticket._creator.lastName} url={ticket._creator.imageUrl}/>
+  <div>{ticket._creator.firstName} {ticket._creator.lastName}</div>
+</div>
+```
+
+### Ticket Page
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-ticket.png" alt="Ticket details" class="img-fluid" figure=false %}
+
+The ticket page allows the user to create or update a ticket. The ticket page is implemented in the `TicketPage.tsx` file.
+Depending on the state of the ticket and the role of the user, different actions are available.
+
+Currently, the following flow is implemented:
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-flow.png" alt="The ticket flow inplemented in FusionDesk" class="img-fluid" figure=false %}
+
+If the ticket is open and the user is an `agent`, the user can mark the ticket as `solved`.
+
+The creator of the ticket then has the option to either close or reopen the ticket with `Accept solution` or `Reject solution`.
+
+After the ticket is closed, it cannot be edited anymore.
+
+### Profile Page
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-profile.png" alt="Profile screen with information about the logged in user" class="img-fluid" figure=false %}
+
+The profile page displays the user information. The profile page is implemented in the `ProfilePage.tsx` file and displays the information of the FusionAuth `user`.
```

Comment:
For each of these pages, it makes sense to show the back end API and then the front end JS.